src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -40,15 +45,35 @@
   public void init() throws ServletException {
     super.init();
     setUserStore(UserStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
   }
 
   /**
-   * Sets the UserStore used by this servlet. This function provides a common setup method for use
-   * by the test framework or the servlet's init() function.
+   * Sets the UserStore, MessageStore, and Conversation Store used by this servlet. This
+   * function provides a common setup method for use by the test framework or the servlet's
+   * init() function.
    */
   void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
+  void setMessageStore(MessageStore messageStore) { this.messageStore = messageStore; }
+  void setConversationStore(ConversationStore conversationStore) { this.conversationStore = conversationStore; }
+
+  /**
+   * Gets the correct stats from datastore and stores them in a map, simulating a json file
+   */
+  private void addStats(HashMap<String, String> map) {
+    // Retrieve sizes of each Datastore
+    Integer userSize = userStore.countTotalUsers();
+    Integer messageSize = messageStore.countTotalMessages();
+    Integer convSize = conversationStore.countTotalConversations();
+
+    // Adds them to the map
+    map.put("Total Users: ", userSize.toString());
```

Comment:
Generally you don't want to hard code user-facing text in the servlet code, that's the job of the jsp. I would make the key either a coded string ("total_users", "total_messages"..) or better yet an enum and let the jsp decide on presentation. Similarly I would not toString() the number, instead leave the value as Integer so the jsp can format. (eg maybe add thousands separators or something). Presentation doesn't belong in the servlet.

Reply:
Thanks a lot, I fixed that, removed the user-facing text from the keys. I left it as a string,string because the next statistics that I'm planning to implement, would be like last message, last conversation created, last user, etc! What do you think?